### PR TITLE
Feat: block migration during snapshot

### PR DESCRIPTION
**Pull request by begonaalvarezd**
# Description of change

The purpose of this PR is to block the user to enter a migration if the snapshot is happening. 

The sensitive areas are `getMigrationData` and `sendMigrationBundle`. The idea is checking the snapshot status each time one of those functions are called, and if snapshot is happening, we display a popup blocking the UI, subscribe to github to wait until the snapshot is over, unblock the user and proceed as normal. 

**Edit: we also block dashboard now**
**Edit2: we also block login now (temp kill switch)**

image

## Links to any relevant issues

N/A

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

To test it, you can change `CHRYSALIS_VARIABLES_ENDPOINT` in `packages/shared/lib/migration.ts` to point to a local gist or any endpoint valid for testing purposes. The response expects a json like:

```json
{
    "snapshot": false
}
```
^^ set to true to simulate a snapshot happening 

Additionally, you need to add your testing endpoint to `packages/desktop/public/index.html` `connect-src 'self'` line to allow the connection.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/shared/lib/migration.ts [hunk 1]
```diff
@@ -143,44 +157,52 @@ export const createMigrationBundle = (inputAddressIndexes: number[], offset: num
...
-                        if (bundle.bundleHash === bundleHash) {
-                            return Object.assign({}, bundle, { migrated: true })
-                        }
-
-                        return bundle
+    return new Promise(async (resolve, reject) => {
+        // Migration: snapshot check
+        await checkChrysalisSnapshot()
+        if (get(ongoingSnapshot) === true) {
+            // we dont display the error because a notification popup will open already
```
Comment (laumair) [inline packages/shared/lib/migration.ts:164]: How come? Say the network stopped while the user was migrating multiple bundles?

File: packages/shared/lib/chrysalis.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (laumair) [inline packages/shared/lib/chrysalis.json:1]: Why do we need this JSON?

File: packages/shared/lib/migration.ts (hunk 1 above)
Comment (begonaalvarezd) [reply packages/shared/lib/migration.ts:164]: We won't display an error notification, a red toast, but we will open the popup for the user and the bundle transaction will still display an error, maybe the message is confusing

File: packages/shared/lib/chrysalis.json (hunk 2 above)
Comment (begonaalvarezd) [reply packages/shared/lib/chrysalis.json:1]: Its the JSON that will be used for the snapshot so it doesnt return a 404, but we can simply remove it and create it when we need it

Comment (obany) [reply packages/shared/lib/chrysalis.json:1]: I would move it to a config folder or similar nearer the root of the repo so its more obvious.

**Review by cvarley100 — APPROVED** [review]

**Review by cvarley100 — APPROVED** [review]

**Pull request merged**